### PR TITLE
For MSAL CPP flows, match exact claims when deleting AT with intersecting scopes, Fixes AB#3095218

### PR DESCRIPTION
**Pull request by sansrivastav**
Related to changes in https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2401
[AB#3095218](https://identitydivision.visualstudio.com/fac9d424-53d2-45c0-91b5-ef6ba7a6bf26/_workitems/edit/3095218)

### CODE REVIEW

File: common4j/src/main/com/microsoft/identity/common/java/cache/MsalCppOAuth2TokenCache.java [hunk 1]
```diff
@@ -148,7 +148,7 @@ public synchronized void saveCredentials(@NonNull final Credential... credential
             }
         }
 
-        saveCredentialsInternal(credentials);
+        saveCredentialsInternal(true, credentials);
```
Comment (nickbopp) [inline common4j/src/main/com/microsoft/identity/common/java/cache/MsalCppOAuth2TokenCache.java]: >saveCredentialsInternal(true [](http://example.com/codeflow?start=8&length=28)

could we bubble this true/false decision up to this method's parameters? This seems risky enough that we'll want to guard it with a flight (even if it starts default-enabled), and I think we'll need that so our layer that understands the flights can make that choice.

Comment (nickbopp) [inline common4j/src/main/com/microsoft/identity/common/java/cache/MsalCppOAuth2TokenCache.java]: >saveCredentialsInternal(true, credentials); [](http://example.com/codeflow?start=8&length=43)

Can we add a test case to common that describes the difference in behavior we expected from true vs. false here?

**Review by nickbopp — APPROVED** [review]
:shipit:

**Review by mohitc1 — APPROVED** [review]
:shipit:

**Pull request merged**